Sources/Moho/Graph.swift

```diff
@@ -0,0 +1,43 @@
+/// Graph.swift
+///
+/// Copyright 2017, The Silt Language Project.
+///
+/// This project is released under the MIT license, a copy of which is
+/// available in the repository.
+
+/// A graph structure capable of answering queries about precedence of keys and
+/// values.
+public final class DAG<K: Comparable & Hashable, T> {
```

Comment:
I think I'd prefer PrecedenceDAG, to avoid conflating this. Also rename the file...